src/components/controls/SearchInput/SearchInput.js

```diff
@@ -43,13 +47,19 @@ export default class SearchInput extends Component {
     onClear: null,
     onSubmit: null,
     theme: 'default',
+    defaultValue: '',
+    value: '',
   }
 
   state = {
-    value: '',
+    value: this.props.value || this.props.defaultValue,
```

Comment:
Como te mencioné antes `null` y `""` son valores explícitos pero en javascript son valores _falsy_, así que terminarían asignando `defaultValue` (incorrectamente).

Debiera estar así (es más verboso, pero más robusto):

```es6
value: this.props.value === undefined
  ? this.props.defaultValue
  : this.props.value
```

`this.props.value || this.props.defaultValue` probablemente funciona igual en este componente porque el valor por defecto es `""` (que es el valor por defecto del `input`), pero en otros componentes podría causar un comportamiento inesperado.